src/ORM/Interfaces/UserInterface.php

```diff
@@ -0,0 +1,12 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Externals\ORM\Interfaces;
+
+/**
+ * Interface to be implemented by any entity
+ * that is acting as an user.
+ */
+interface UserInterface
```

Comment:
I feel like there should be an interface requirement for _some_ unique ID field here. I suspect most implementations of course just wrap it on to something else.

Reply:
added a `getId` on interface, trying to keep the definition same as `EntityInterface`

Reply:
`getId` is problematic, having clashes of function names across interfaces poses problems where what if you wanted to use a different id for this interface than the one required by EntityInterface?

I'd prefer to see it named `getAuthId()` or something else.

Reply:
We're also going to need to add a changelog to externals for this change so that when someone wants to upgrade to 2.0 they know what needs to be done (which is an issue for the other changes but we can do them in another ticket)

Reply:
Totally happy with `getAuthId()`. having `public function getAuthId() { return $this->getId(); }` is completely fine.

Reply:
added `getUniqueId()` and also created a changelog